open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,509 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.wso2.openbanking.accelerator.consent.extensions.validate.impl;
+
+import com.wso2.openbanking.accelerator.common.util.ErrorConstants;
+import com.wso2.openbanking.accelerator.consent.extensions.common.ConsentExtensionConstants;
+import com.wso2.openbanking.accelerator.consent.extensions.validate.util.ConsentValidatorUtil;
+import net.minidev.json.JSONObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Class for validating VRP submission request.
+ */
+public class VRPSubmissionPayloadValidator {
+
+    private static final Log log = LogFactory.getLog(VRPSubmissionPayloadValidator.class);
+
+    /**
+     * Validates the initiation parameters between the initiation of submission request and the initiation parameters
+     * of consent initiation request.
+     *
+     * @param initiationOfSubmission                 The initiation parameters from the submission request.
+     * @param initiationParameterOfConsentInitiation The initiation parameters from the consent initiation request.
+     * @return A JSONObject indicating the validation result. It contains a boolean value under the key
+     * ConsentExtensionConstants.IS_VALID_PAYLOAD, indicating whether the payload is valid. If the
+     * validation fails, it returns a JSONObject containing error details with keys defined in ErrorConstants.
+     */
+    public static JSONObject validateInitiation(JSONObject initiationOfSubmission,
+                                                JSONObject initiationParameterOfConsentInitiation) {
+
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
```

Comment:
https://github.com/wso2/financial-open-banking/pull/58/files#r1618419414 comment applies to hear as well

Reply:
Fixed in [8362873]